rsc-server/src/org/openrsc/net/codec/PacketEncoder.java

```diff
@@ -0,0 +1,36 @@
+package org.openrsc.net.codec;
+
+import org.jboss.netty.channel.Channel;
+import org.jboss.netty.buffer.ChannelBuffer;
+import org.jboss.netty.buffer.ChannelBuffers;
+import org.jboss.netty.channel.ChannelHandlerContext;
+import org.jboss.netty.handler.codec.oneone.OneToOneEncoder;
+import org.openrsc.net.packet.Packet;
+
+/**
+ * Writes an outgoing packet.
+ */
+public class PacketEncoder extends OneToOneEncoder {
+
+	@Override
+	protected Object encode(ChannelHandlerContext ctx, Channel channel, Object obj) throws Exception {
+		// Get the packet object.
+		Packet packet = (Packet) obj;
+		
+		 // Create a new buffer.
+		ChannelBuffer buffer = ChannelBuffers.dynamicBuffer();
+		
+		// Write the packet length.
+		buffer.writeInt(packet.getPacketLength());
+		
+		// Write the packet opcode.
+		buffer.writeInt(packet.getOpcode() & 0xff);

```

Comment:
To prevent the opcode value overflowing 1 byte size

Example:
https://gist.github.com/acrois/e2291313c249837148adb477bd186b29

The question is, does this value need to be limited to 0-255 because it is being sent across the wire using 4 bytes? The PacketDecoder is working in the same fashion, to prevent the opcode value being larger than 255.

From PacketDecoder.java:
```
	// Read the opcode.
	final int opcode = buffer.readInt() & 0xff;
```

Reply:
Hey @acrois, thanks for your comment! I've rolled back this server for now while I focus on the client, as CMorgue has moved onto other projects.